### PR TITLE
add match- Co-authored-by: Alexa

### PR DESCRIPTION
**Pull request by hannah68**

### CODE REVIEW

**Review by Can-Tina — APPROVED** [review]

**Review by TheunisK — APPROVED** [review]

File: src/client/pages/matchPage/MatchPage.jsx [hunk 1]
```diff
@@ -1,53 +1,118 @@
...
+    const [book, setBook] = useState({});
+    const [displayMatchCard, setDisplayMatchCard] = useState(true);
+    const [recommendations, setRecommendations] = useState([]);
+
+    const pickRandomBook = (library) => {
+        const generatedIndex = generateRandomInt(library.length - 1);
+        
+        return library[generatedIndex];
+    }
+    console.log(library);
```
Comment (alexamaingard) [inline src/client/pages/matchPage/MatchPage.jsx:21]: Remove printing statement :)

File: src/client/pages/matchPage/MatchPage.jsx [hunk 2]
```diff
@@ -1,53 +1,118 @@
...
+    const pickGroupFromDifferentGroup = (unwantedGroupId) => {
+        let bookToMatch;
+
+        do{
+            bookToMatch = pickRandomBook(library);
+        } while (bookToMatch.groupId === unwantedGroupId);
```
Comment (vherus) [inline src/client/pages/matchPage/MatchPage.jsx:27]: The rare and fabled `do while`! Nice

File: src/client/pages/matchPage/MatchPage.jsx [hunk 3]
```diff
@@ -1,53 +1,118 @@
...
 
 const MatchPage = () => {
-    // const [book, setBook] = useState({});
+    const [library, setLibrary] = useState({});
+    const [book, setBook] = useState({});
+    const [displayMatchCard, setDisplayMatchCard] = useState(true);
+    const [recommendations, setRecommendations] = useState([]);
+
+    const pickRandomBook = (library) => {
+        const generatedIndex = generateRandomInt(library.length - 1);
```
Comment (vherus) [inline src/client/pages/matchPage/MatchPage.jsx:17]: What happens here if `library.length` is 0? You'd effectively be doing `generateRandomInt(-1)` - how would that affect your app?

**Review by vherus — APPROVED** [review]

File: src/client/pages/matchPage/MatchPage.jsx (hunk 2 above)
Comment (alexamaingard) [reply src/client/pages/matchPage/MatchPage.jsx:27]: Thanks. It took me a sec to remember the syntax :')

File: src/client/pages/matchPage/MatchPage.jsx (hunk 3 above)
Comment (alexamaingard) [reply src/client/pages/matchPage/MatchPage.jsx:17]: You have a strong point here. Will modify

**Pull request merged**